_posts/2019-08-12-what-the-cluster.markdown

```diff
@@ -0,0 +1,75 @@
+
+### Cluster Keys
+
+Used to help keep all frequently accessed data in the same micro-partition for faster retrieval. This is only ideal for very large tables as snowflake automatically clusters your data into micro-partitions . Using cluster keys will override snowflakes natural clustering.
+So say you want to find unicorns; you want to know how old they are, how big they are and where they were last spotted. This is the information you need to go forth and find unicorns so you want this information to be stored in the same micro-partition so you can get hold of this vital information fast.
+Cluster keys are used to keep frequently accessed data in the same micro-partition.So if I wanted to re cluster my UNICORNS table from above I could set clustering keys on AGE,SIZE and LOCATION.
+
+```sql
+ALTER TABLE UNICORNS CLUSTER BY (AGE,SIZE,LOCATION);
+ALTER TABLE UNICORNS RECLUSTER;
+```
+
+### Multi Cluster Warehouses
+
+Multi cluster warehouses can automatically scale out and spin up more compute clusters based on user concurrency needs. With multi cluster warehouses you can configure the minimum and maximum number of server clusters up to a maximum of ten. Snowflake documentation is really good and you can find out how much multi cluster warehouses will cost you in credits in different scenarios [here:](https://docs.snowflake.net/manuals/user-guide/warehouses-multicluster.html)
+The two scaling strategies allow you to run using maximised clusters, when you start your warehouse it will automatically use all the clusters.Alternatively you can use the auto scaling strategy so that clusters will be started up only when they are needed.
+
+Setting up your warehouse is as simple as: 
+```sql
+CREATE OR REPLACE WAREHOUSE UNICORN_WH
+WAREHOUSE_SIZE = 'SMALL'
+AUTO_RESUME = TRUE
+MIN_CLUSTER_COUNT = 1
+MAX_CLUSTER_COUNT = 5
+INITIALLY_SUSPENDED = TRUE
+COMMENT = 'unicorn finder';
+```
+
+### Wrapping up
+
+Snowflake is a fast and powerful data warehousing solution with a lot of terminology to get your head around. Once you do you can start designing your own solution, do you need a single compute cluster or a multi cluster warehouse? Now that you know about all the different Snowflake clustering terms, you can start Snowflaking for yourself.Still not sure? We would be more than happy to help you.
+
+Go forth and cluster.
+
+If you think we can help you cluster unicorns in Snowflake, feel free to [contact-us](https://www.mechanicalrock.io/lets-get-started)
+
+### References
+
+[https://docs.snowflake.net/manuals/user-guide/warehouses-multicluster.html](https://docs.snowflake.net/manuals/user-guide/warehouses-multicluster.html)
+[https://docs.snowflake.net/manuals/user-guide/intro-key-concepts.html](https://docs.snowflake.net/manuals/user-guide/intro-key-concepts.html)
```

Comment:
these two will be jammed up right against each other, on the same line

Maybe a dot-list would be better